### PR TITLE
simplify CI setup

### PR DESCRIPTION
**Pull request by akash1810**
## What does this change?

Adds all build steps into VCS to make it more explicit and to allow branches to update CI independently of each other.

Worth noting:
- using the sbt runner directly in TeamCity is not advised as it always (or certainly used to always) exits 0, which leads to confusing results
- the nvm runner on TeamCity is no longer working, invoking nvm commands ourselves resolves this

## How can we measure success?

Simplified CI setup which is fully in the control of, and tracked in, the repository being built.

## Have we considered potential risks?

n/a

## Images

n/a

### CODE REVIEW

**Comment by SHession** [comment]
Oh no, build failed. I approved too quickly!

**Comment by akash1810** [comment]
> Need to makes the changes to TC I guess?

Yep! I'll do that now and trigger a new build. If it goes green will merge.

File: teamcity.sh [hunk 1]
```diff
@@ -2,5 +2,34 @@
...
+
+  setupNvm
+
+  # clear old packages first
+  rm -rf node_modules
+
+  npm install
+  npm run build
+
+  echo "##teamcity[compilationStarted compiler='js']"
```
Comment (SHession) [inline teamcity.sh:25]: I assume this should be compilationFinished?

Comment (akash1810) [reply teamcity.sh]: Ah yes! Great spot. Fixed now.

File: teamcity.sh [hunk 2]
```diff
@@ -2,5 +2,34 @@
...
+
+  npm install
+  npm run build
+
+  echo "##teamcity[compilationFinished compiler='js']"
+}
+
+buildSbt() {
+  echo "##teamcity[compilationStarted compiler='sbt']"
+  sbt clean test scripts/compile riffRaffUpload
```
Comment (SHession) [inline teamcity.sh:30]: Should `scripts/compile` be `compile`?

Comment (SHession) [reply teamcity.sh:30]: Sorry for reviewing this in an awkward way!

Comment (akash1810) [reply teamcity.sh:30]: Aw, I thought I corrected this before you spotted it! Yes it should!

Comment (akash1810) [reply teamcity.sh:30]: This should be done now. At least I've force pushed it, but GitHub is taking a little while to see it.

**Pull request merged**